src/Adapter/Cart/QueryHandler/GetCartSummaryHandler.php

```diff
@@ -0,0 +1,206 @@
+
+                $product['image_link'] = Context::getContext()->link->getImageLink(
+                    $product['link_rewrite'],
+                    $product['id_image'],
+                    'small_default'
+                );
+
+                if (!isset($product['attributes_small'])) {
+                    $product['attributes_small'] = '';
+                }
+
+                $product['customized_datas'] = Product::getAllCustomizedDatas(
+                    (int) $cart->id,
+                    null,
+                    true,
+                    null,
+                    (int) $product['id_customization']
+                );
+            }
+        }
+
+        if (count($summary['discounts'])) {
+            foreach ($summary['discounts'] as &$voucher) {
+                $voucher['value_real'] = Tools::displayPrice($voucher['value_real'], $currency);
+            }
+        }
+
+        if (isset($summary['gift_products']) && count($summary['gift_products'])) {
+            foreach ($summary['gift_products'] as &$product) {
+                $product['image_link'] = Context::getContext()->link->getImageLink(
+                    $product['link_rewrite'],
+                    $product['id_image'],
+                    'small_default'
+                );
+
+                if (!isset($product['attributes_small'])) {
+                    $product['attributes_small'] = '';
+                }
+            }
+        }
+
+        return $summary;
+    }
+
+    private function getDeliveryOptions(Cart $cart)
+    {
+        $deliveryOptionListFormatted = [];
+        $deliveryOptionList = $cart->getDeliveryOptionList();
+
+        if (!count($deliveryOptionList)) {
+            return [];
+        }
+
+        $defaultCarrierId = (int) Configuration::get('PS_CARRIER_DEFAULT');
+
+        foreach (current($deliveryOptionList) as $key => $deliveryOption) {
+            $name = '';
+            $first = true;
+            $idDefaultCarrierDelivery = false;
+
+            foreach ($deliveryOption['carrier_list'] as $carrier) {
+                if (!$first) {
+                    $name .= ', ';
+                } else {
+                    $first = false;
+                }
+
+                $name .= $carrier['instance']->name;
+
+                if ($deliveryOption['unique_carrier']) {
+                    $name .= ' - ' . $carrier['instance']->delay[Context::getContext()->employee->id_lang];
+                }
+
+                if (!$idDefaultCarrierDelivery) {
+                    $idDefaultCarrierDelivery = (int) $carrier['instance']->id;
+                }
+
+                if ((int) $carrier['instance']->id === $defaultCarrierId) {
+                    $idDefaultCarrierDelivery = $defaultCarrierId;
+                }
+
+                if (!$cart->id_carrier) {
+                    $cart->setDeliveryOption([$cart->id_address_delivery => (int) $carrier['instance']->id . ',']);
+                    $cart->save();
```

Comment:
even though it's `QueryHandler` it performs write operation. :/ it feels wrong, but this is part of legacy behavior, so i think its not worth moving out, wdyt?

Reply:
Indeed you cant change the legacy behavior :/ at least you can put a warning in comment, along with a `@todo`